pkg/controller/operators/olm/apiservices.go

```diff
@@ -921,3 +978,53 @@ func (a *Operator) isAPIServiceAdoptable(target *v1alpha1.ClusterServiceVersion,
 	adoptable = ownerutil.AdoptableLabels(apiService.GetLabels(), true, owners...)
 	return
 }
+
+// deleteLegacyAPIServiceResources deletes resources that were created by OLM for an APIService that used the old naming convention.
+func (a *Operator) deleteLegacyAPIServiceResources(namespace string, desc v1alpha1.APIServiceDescription) error {
```

Comment:
everything here should probably do a `get` from the lister to check that these resources are owned by OLM before delete.

Reply:
What is the correct way to ensure that the resources are owned by OLM?

Reply:
A check for a CSV OwnerReference or OwnerLabel?

Reply:
Yep exactly. I think there's already some code in this package for checking if a resource is adoptable.